components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/OptimizedApplicationConfig.java

```diff
@@ -136,13 +181,13 @@ public List<String> getLocalAuthenticatorConfigNames() {
         }
 
         /**
-         * This method is used to get the resource ids of the federated IdPs of authentication step.
+         * This method is used to get the Optimized federated IDPs of authentication step.
          *
          * @return List of resource ids.
          */
-        public List<String> getFederatedIdPResourceIds() {
+        public List<OptimizedFederatedIdP> getOptimizedFederatedIdPs() {
```

Comment:
Does not it cause method not found exception if change the public method signature.

Reply:
+1. we can't remove the existing public methods, let's keep the existing method and mark as deprecated. In the implementation, we can return IDP ids from optimizedFederatedIdPs

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/commit/87aac03e1fe4a099ff0d87cfced2667a1a3f74ad